jenkins/run_tests.sh

```diff
@@ -20,12 +34,38 @@ pushd ansible-osbuild
     playbook.yml
 popd
 
+# Modify the URLs in the test cases to use a nearby mirror.
+TEST_CASE_DIR=/usr/share/tests/osbuild-composer
+FASTEST_MIRROR=$(get_fastest_mirror)
+sed -i "s#http://dl.fedoraproject.org/pub/fedora/linux/releases/#${FASTEST_MIRROR}releases/#" ${TEST_CASE_DIR}/cases/*.json
+
+# Test wrapper that automatically collects logs.
 run_osbuild_test() {
   TEST_NAME=$1
+  TEST_CASE=${2:-}
+
+  # Log each test case individually, if provided.
+  if  [[ -z $TEST_CASE ]]; then
+    LOGFILE=${TEST_NAME}.log
+  else
+    # Strip the test case name down to just the filename without extension.
+    FILENAME=$(basename $TEST_CASE)
+    LOGFILE=${TEST_NAME}-${FILENAME%%.*}.log
+  fi
+
   /usr/libexec/tests/osbuild-composer/${TEST_NAME} \
-    -test.v | tee ${TEST_NAME}.log > /dev/null &
+    -test.v $TEST_CASE 2>&1 | tee $LOGFILE > /dev/null &
 }
 
+# Try an upload to AWS.
+if [[ $NAME == 'Fedora' ]] && [[ $VERSION_ID == "31" ]]; then
+  run_osbuild_test osbuild-image-tests \
+    ${TEST_CASE_DIR}/cases/fedora_31-x86_64-ami-boot.json
+  wait
```

Comment:
Why do you start the osbuild-image executable in the background and then wait for it? Why not execute it in foreground ?